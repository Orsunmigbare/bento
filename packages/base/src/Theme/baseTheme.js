import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
