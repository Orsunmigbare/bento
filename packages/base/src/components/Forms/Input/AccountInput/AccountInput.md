```jsx harmony
<AccountInput value={'34951515827'}/>
```

```jsx harmony
<AccountInput/>
```