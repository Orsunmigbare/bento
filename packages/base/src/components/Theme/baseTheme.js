export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
