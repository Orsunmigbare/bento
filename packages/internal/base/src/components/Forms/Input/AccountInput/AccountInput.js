import React from "react"
import Input, { InputPropTypes } from "../Input"
import { accountMasks } from "../masks"

const AccountInput = ({ mask, locale = "nb", ...otherProps }) => (
  <Input type={"tel"} mask={accountMasks[locale]} {...otherProps} />
)

AccountInput.propTypes = InputPropTypes

export default AccountInput
