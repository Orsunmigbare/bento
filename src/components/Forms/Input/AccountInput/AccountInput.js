import React, { useRef } from "react"
import PropTypes from "prop-types"
import Input, { InputPropTypes } from "../Input"
import { accountMasks } from "../masks"

const AccountInput = React.forwardRef(
  ({ mask, locale = "nb", ...otherProps }, ref) => {
    const inputRef = useRef(ref)
    return (
      <Input
        type="tel"
        ref={inputRef}
        options={accountMasks[locale]}
        {...otherProps}
      />
    )
  }
)

AccountInput.propTypes = {
  ...InputPropTypes,
  locale: PropTypes.oneOf(["nb"]),
  onChange: PropTypes.func,
  id: PropTypes.string.isRequired,
  guide: PropTypes.bool
}

AccountInput.defaultProps = {
  locale: "nb",
  guide: false
}

export default AccountInput
