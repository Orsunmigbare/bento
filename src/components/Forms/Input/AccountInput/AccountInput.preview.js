import React from "react"
import AccountInput from "./AccountInput"
import Toggle from "../../Toggle/Toggle"
import Label from "../../Label/Label"

const preview = {
  title: "AccountInput",
  category: "Components/Forms/Input",
  component: AccountInput,
  render: props => (
    <AccountInput label={"AccountInput"} id="9852145" {...props} />
  ),
  renderExampleController: ({ setComponentState }) => (
    <React.Fragment>
      <Label for="guideAccountInput">Toggle guides</Label>
      <Toggle
        id="guideAccountInput"
        onChange={e => setComponentState({ guide: e.target.checked })}
      >
        Guide
      </Toggle>
    </React.Fragment>
  )
}

export default preview
