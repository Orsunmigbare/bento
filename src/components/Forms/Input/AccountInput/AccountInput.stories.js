import React from "react"
import { storiesOf } from "@storybook/react"
import AccountInput from "./AccountInput"

storiesOf("components|base/Forms/Input/AccountInput", module).add(
  "no props",
  () => {
    return <AccountInput label={"AccountInput"} id="9852145" />
  }
)
