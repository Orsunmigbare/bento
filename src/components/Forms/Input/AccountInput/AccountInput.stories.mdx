import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import AccountInput from "./AccountInput"

<Meta title="Components|Forms/Input/AccountInput" component={AccountInput}/>


# AccountInput
This components controlls user input so that they can only enter numbers with the format of an accountNumber



```javascript
import {AccountInput} from "@staccx/bento"
```


## Standard AccountInput
The inputfield only allowed numbers and takes number in a 4-2-5 format with seperators between and max 11 carrecters


<Preview>
  <Story name="Standard">
   <AccountInput label={"AccountInput"} id="9852145" />
  </Story>
</Preview>





## Props for AccountInput
<Props of={AccountInput} />
