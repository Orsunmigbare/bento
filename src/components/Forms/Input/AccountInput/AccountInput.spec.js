import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../../theming/themes/baseTheme"
import AccountInput from "./AccountInput"
import "@testing-library/jest-dom"

const setup = () => {
  const utils = render(
    <ThemeProvider theme={baseTheme}>
      <AccountInput id="1" label="input" />
    </ThemeProvider>
  )
  const input = utils.getByLabelText("input")
  return {
    input,
    ...utils
  }
}

describe("AccountInput", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <AccountInput id="wegwf" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Type", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <AccountInput id="wegwf" type="text" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it("Label", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <AccountInput id="wegwf" label="Balance" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("simulate input", () => {
      const { input } = setup()
      fireEvent.change(input, { target: { value: 11018420038 } })
      expect(input.value).toBe("11018420038")
    })
  })
})

describe("Rendering", () => {
  it("Should render", () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <AccountInput id="test" data-testid="testID" label="Balance" />
      </ThemeProvider>
    )
    expect(screen.getByTestId("testID")).toBeInTheDocument()
  })
})
