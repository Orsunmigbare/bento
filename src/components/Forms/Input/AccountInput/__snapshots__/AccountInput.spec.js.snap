// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInput Snapshots No props 1`] = `
<div
  className="sc-AykKI iDYrmt"
>
  
  
  <input
    autoFocus={null}
    className="sc-AykKJ gjiytr"
    disabled={false}
    hidden={false}
    id="wegwf"
    name={null}
    onBlur={null}
    onChange={null}
    onKeyDown={null}
    placeholder={null}
    style={null}
    type="tel"
  />
</div>
`;
